src/esssans/i_of_q.py

```diff
@@ -151,17 +158,28 @@ def merge_spectra(
     :
         The input data converted to Q and then summed over all detector pixels.
     """
+    if final_dims is None:
+        final_dims = ['Q']
+    if (wavelength_bands is not None) and (wavelength_bands.ndim == 1):
+        # We expect wavelength_bands to be two-dimensional below
+        wavelength_bands = wavelength_bands.fold(
+            dim='wavelength', sizes={**{'band': 1}, **wavelength_bands.sizes}
+        )
```

Comment:
Yes, I wanted to be able to define the bands via a single variable.

The variable can be 2d (e.g. 10 bands, with a start and end wavelength for each band, so the shape would be `{'band: 10, 'wavelength': 2}`), which allows for overlapping bands.

For convenience, the variable could also be just 1d, a single array of non-overlapping bands (which is a common use case). In that case, I would automatically fold it to a 2d array here,

Reading what I just wrote above,
1. I think this piece of code is wrong, it doesn't do what I want it to do, just adds a dummy dim when it should be making something that has shape `{'band: wavelength_bands.sizes, 'wavelength': 2}`.
2. Because I had to explain all this, I either need to come up with a better interface, e.g. ask for 2 arrays with start and end wavelengths (in that case, there would be a lot of duplicates in the case of non-overlapping bands), or at the very least document it much better in the docstrings.

Reply:
See also my comments on performance. I think unless overlapping bands are very common we may want to handle this differently anyway, to avoid the performance issues?